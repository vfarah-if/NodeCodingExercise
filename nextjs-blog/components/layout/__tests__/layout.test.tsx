import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Layout from '..';

describe('Layout', () => {
  test('should generate a home page with expected home layout', () => {
    const component = render(
      <Layout isHome={true}>
        <section>
          <p>
            Having worked with Next JS without doing any tutorials, I thought I
            would go back and take nothign for granted.
          </p>
        </section>
      </Layout>
    );

    expect(component.container).toMatchSnapshot('HomeLayout');
  });

  test('should generate a detail different to home layout', () => {
    const component = render(
      <Layout isHome={false}>
        <section>
          <p>
            Having worked with Next JS without doing any tutorials, I thought I
            would go back and take nothign for granted.
          </p>
        </section>
      </Layout>
    );

    expect(component.container).toMatchSnapshot('NotHomeLayout');
  });
});
