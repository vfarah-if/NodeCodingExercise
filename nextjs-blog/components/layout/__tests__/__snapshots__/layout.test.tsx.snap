// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should generate a detail different to home layout: NotHomeLayout 1`] = `
<div>
  <div>
    <header>
      <a
        href="/"
      >
        <img
          alt="Jon Doe"
          class="undefined undefined"
          src="/images/profile.jpg"
        />
      </a>
      <h2>
        <a
          href="/"
        >
          Jon Doe
        </a>
      </h2>
    </header>
    <main>
      <section>
        <p>
          Having worked with Next JS without doing any tutorials, I thought I would go back and take nothign for granted.
        </p>
      </section>
    </main>
    <div>
      <a
        href="/"
      >
        ‚Üê Back to home
      </a>
    </div>
  </div>
</div>
`;

exports[`Layout should generate a home page with expected home layout: HomeLayout 1`] = `
<div>
  <div>
    <header>
      <img
        alt="Jon Doe"
        class="undefined undefined"
        src="/images/profile.jpg"
      />
      <h1>
        Jon Doe
      </h1>
    </header>
    <main>
      <section>
        <p>
          Having worked with Next JS without doing any tutorials, I thought I would go back and take nothign for granted.
        </p>
      </section>
    </main>
  </div>
</div>
`;
